File: Wifi Range.cpp
class Solution{
    public:
    bool wifiRange(int N, string s, int X){
        for(int i=0; i<N; i++){
            if(s[i] == '1'){
                int k = 1;
                while(i-k >= 0 && k<=X){
                    if(s[i-k] == '0')
                        s[i-k] = '2'; //such that we can differentiate between the real room with wifi
                        // and room with range of wifi with anearby rooms
                    k++;
                }
                k = 1;
                while(i+k < N && k<=X){
                    if(s[i+k] == '0')
                        s[i+k] = '2';
                    k++;
                }
            }
        }
        for(int i=0; i<N; i++){
            if(s[i] == '0'){
                return 0;
            }
        }
        return 1;
    }
};

File: Wifi Range.java
class Solution 
{ 
    boolean wifiRange(int N, String S, int X) 
    { 
        // code here
        StringBuilder sb = new StringBuilder();
        sb.append(S);
        int i = 0;
        while(i<N){
            if(S.charAt(i) == '1'){
                int range = X;
                i++;
                while(i<N && range>0 && S.charAt(i) != '1'){
                    sb.setCharAt(i,'1');
                    i++;
                    range--;
                }
            }
            else
            i++;
        }
        i = N-1;
        while(i>=0){
            if(S.charAt(i) == '1'){
                int range = X;
                i--;
                while(i>=0 && range>0 && S.charAt(i) != '1'){
                    sb.setCharAt(i,'1');
                    i--;
                    range--;
                }
            }
            else
            i--;
        }
        if(sb.toString().contains("0"))
        return false;
        return true;
    }
} 
